pageserver/src/tenant/layer_map.rs

```diff
@@ -473,18 +543,25 @@ where
     /// TODO The optimal number should probably be slightly higher than 1, but to
     ///      implement that we need to plumb a lot more context into this function
     ///      than just the current partition_range.
-    pub fn is_reimage_worthy(layer: &L, partition_range: &Range<Key>) -> bool {
+    pub fn is_reimage_worthy(
+        layer: &L,
+        partition_range: &Range<Key>,
+        ctx: &RequestContext,
+    ) -> Result<bool> {
+        if !layer.overlaps(partition_range, ctx)? {
+            return Ok(false);
+        }
```

Comment:
Please update the comment above, for the addition of this case

Reply:
I'm a little surprised the layer.overlaps function is needed at all. We store the holes in the layer map, and use them during searches to avoid visiting layers unnecessarily, right? Could we use the information from the layer map here, too?

Reply:
`overlaps` was originally introduced in this PR and used in `LayerMap::search`. 
Right now it seems to be really not needed, because counting stacked deltas is based on stored key ranges, not on key range of layer itself. @bojanserafimov  cam you confirm it?

Reply:
> counting stacked deltas is based on stored key ranges, not on key range of layer itself.

Thanks for checking, because this is false :) I think this should be the only place that needs change: https://github.com/neondatabase/neon/blob/58fa4f0eb7f237e5d327566819eb4317b951554f/pageserver/src/tenant/layer_map.rs#L608

Instead of iterating layers, here we should be iterating occupied ranges of all layers.

But yes, I don't see a need for `layer.overlaps`.